akka-cluster/src/multi-jvm/scala/akka/cluster/RemoteFeaturesWithWatcherClusterWatcheeNoClusterSpec.scala

```diff
@@ -0,0 +1,198 @@
+    extends ClusterWatcherNoClusterWatcheeSpec(new ClusterWatcherNoClusterWatcheeConfig(useUnsafe, artery = true))
+
+private object ClusterWatcherNoClusterWatcheeSpec {
+  final case class WatchIt(watchee: ActorRef)
+  case object Ack
+  final case class WrappedTerminated(t: Terminated)
+
+  class Listener(testActor: ActorRef) extends Actor {
+    def receive: Receive = {
+      case WatchIt(watchee) =>
+        context.watch(watchee)
+        sender() ! Ack
+      case t: Terminated =>
+        testActor.forward(WrappedTerminated(t))
+    }
+  }
+}
+
+abstract class ClusterWatcherNoClusterWatcheeSpec(multiNodeConfig: ClusterWatcherNoClusterWatcheeConfig)
+    extends MultiNodeSpec(multiNodeConfig)
+    with MultiNodeClusterSpec
+    with ImplicitSender
+    with ScalaFutures {
+
+  import ClusterWatcherNoClusterWatcheeSpec._
+  import multiNodeConfig._
+
+  override def initialParticipants: Int = roles.size
+
+  muteDeadLetters(Heartbeat.getClass)()
+
+  protected val probe = TestProbe()
+
+  protected def identify(role: RoleName, actorName: String, within: FiniteDuration = 10.seconds): ActorRef =
+    identifyWithPath(role, "user", actorName, within)
+
+  protected def identifyWithPath(
+      role: RoleName,
+      path: String,
+      actorName: String,
+      within: FiniteDuration = 10.seconds): ActorRef = {
+    system.actorSelection(node(role) / path / actorName) ! Identify(actorName)
+    val id = expectMsgType[ActorIdentity](within)
+    assert(id.ref.isDefined, s"Unable to Identify actor [$actorName] on node [$role].")
+    id.ref.get
+  }
+
+  private val provider = RARP(system).provider
+
+  s"Remoting with UseUnsafeRemoteFeaturesWithoutCluster enabled=$useUnsafe, " +
+  "watcher system using `cluster`, but watchee system using `remote`" must {
+
+    val send = if (system.settings.HasCluster || (!system.settings.HasCluster && useUnsafe)) "send" else "not send"
+
+    s"$send `Watch`/`Unwatch`/`Terminate` when watching from cluster to non-cluster remoting watchee" in {
+      runOn(remoting) {
+        system.actorOf(Props(classOf[Listener], probe.ref), "watchee")
+        enterBarrier("watchee-created")
+        enterBarrier("watcher-created")
+      }
+
+      runOn(clustered) {
+        enterBarrier("watchee-created")
+        val watcher = system.actorOf(Props(classOf[Listener], probe.ref), "watcher")
+        enterBarrier("watcher-created")
+
+        val watchee = identify(remoting, "watchee")
+        probe.send(watcher, WatchIt(watchee))
+        probe.expectMsg(1.second, Ack)
+        provider.remoteWatcher.get ! Stats
+        awaitAssert(expectMsgType[Stats].watchingRefs == Set((watchee, watcher)), 2.seconds)
+      }
+      enterBarrier("cluster-watching-remote")
+
+      runOn(remoting) {
+        system.stop(identify(remoting, "watchee"))
+        enterBarrier("watchee-stopped")
+      }
+
+      runOn(clustered) {
+        enterBarrier("watchee-stopped")
+        probe.expectMsgType[WrappedTerminated](2.seconds)
```

Comment:
this is wrong when `useUnsafe=false`, then it shouldn't allow watch from `clustered` to `remoting`, and the reason it wrongly works now is the `RemoteWatcher.shouldWatch` as mentioned above

Reply:
Right, that was my misunderstanding of use unsafe not applying when originating from a cluster node vs a remote only.

Reply:
Fixed